applications/eradius_prometheus_collector/src/eradius_prometheus_collector.erl

```diff
@@ -252,18 +258,65 @@ get_metric_info(Name, Stat) ->
         server -> {Metrics, proplists:get_value(Metric, NasFields), RadiusMetricType}
     end.
 
-%% Helper to get a value of a server/nas metric by the given #nas_counter{} index
+%% Helper to get a value of a server/nas metric by the given #nas_counter{} or #client_counter{} index
 %% @private
-get_server_labels_and_val(MetricIdx, #nas_counter{} = Cnt) ->
+get_labels_and_val(_, {#nas_counter{} = Cnt, server}, {Name, Labels})
+  when Name =:= accounting_requests_total;
+       Name =:= accounting_responses_total ->
+    Type = proplists:get_value(acct_type, Labels),
+    {{ServerIP, ServerPort}, NasIP, NasId} = Cnt#nas_counter.key,
+    case lists:member(Type, ?ACCT_TYPES) of
+        true ->
+            Value = mapping_accounting(Name, Type, Cnt),
+            {Value, [{server_name, Cnt#nas_counter.server_name}, {server_ip, inet:ntoa(ServerIP)}, {server_port, ServerPort},
+                     {nas_id, NasId}, {nas_ip, inet:ntoa(NasIP)}, {acct_type, Type}]};
+        false ->
+            lists:map(fun (AcctType) ->
+                Value = mapping_accounting(Name, AcctType, Cnt),
+                {Value, [{server_name, Cnt#nas_counter.server_name}, {server_ip, inet:ntoa(ServerIP)}, {server_port, ServerPort},
+                         {nas_id, NasId}, {nas_ip, inet:ntoa(NasIP)}, {acct_type, AcctType}]}
+            end, ?ACCT_TYPES)
+    end;
+get_labels_and_val(MetricIdx, {#nas_counter{} = Cnt, server}, _) ->
     {{ServerIP, ServerPort}, NasIP, NasId} = Cnt#nas_counter.key,
     Value = element(MetricIdx + 1, Cnt),
     {Value, [{server_name, Cnt#nas_counter.server_name}, {server_ip, inet:ntoa(ServerIP)}, {server_port, ServerPort},
-             {nas_id, NasId}, {nas_ip, inet:ntoa(NasIP)}]}.
+             {nas_id, NasId}, {nas_ip, inet:ntoa(NasIP)}]};
+get_labels_and_val(_, {#client_counter{} = Cnt, client}, {Name, Labels})
+  when Name =:= client_accounting_requests_total; 
+       Name =:= client_accounting_responses_total ->
+    Type = proplists:get_value(acct_type, Labels),
+    {{ClientName, ClientIP, _ClientPort}, {_, ServerIP, ServerPort}} = Cnt#client_counter.key,
+    case lists:member(Type, ?ACCT_TYPES) of
```

Comment:
checking type against ACCT_TYPES is a bit excessive. acct_type is either set and valid, or not set at all. Checking the return for proplists:get_value for undefined should be sufficient.